### ISSUE TITLE
salt-ssh not properly syncing pillar data

### ISSUE THREAD

**Issue by SaschaSchuenemann** [saltstack/salt#27620]
When running salt-ssh the pillar data seems to get wrongly synced when calling execution modules.

``` salt
ccm-asm:/srv/salt # salt ubu-lxc --out yaml pillar.item my_data
ubu-lxc:
  my_data:
    test_string: hello
```

``` salt
ccm-asm:/srv/salt # salt ubu-lxc --out yaml my_test.pillar my_data
ubu-lxc:
  my_data:
    test_string: hello
```

``` salt
ccm-asm:/srv/salt # salt-ssh ubu-lxc --out yaml pillar.item my_data
ubu-lxc:
  my_data:
    test_string: hello
```

``` salt
ccm-asm:/srv/salt # salt-ssh ubu-lxc --out yaml my_test.pillar my_data
ubu-lxc: {}
```

I have a gist example at: https://gist.github.com/SaschaSchuenemann/628d3f16166ad9d0eeea

The problem occurs in at least two versions:
salt 2015.5.5 (Lithium)
salt 2015.8.0 (Beryllium)

**Comment by SaschaSchuenemann** [comment]
Extension to this issue:

Calling a custom state module via a state formula seems to use a different pillar than when the same module is called from within a custom state execution module

**Comment by throwawayj** [comment]
I am havin ghte same issue as SaschaSchuenemann  on version 2015.5.5

**Comment by jfindlay** [comment]
@SaschaSchuenemann, @throwawayj, thanks for the report.  @SaschaSchuenemann, does your second comment also only apply to salt-ssh?

**Comment by SaschaSchuenemann** [comment]
@jfindlay yes, I have encountered this only with salt-ssh.

It seems that the calls to wfunc(s) (like pillar.*, saltutils.* ) are handled different than calls to custom modules. Our guess is that not only pillar data but other data like grains and mines might also be affected.

We were able to properly access pillar data using salt-ssh when placing a wrapper in /usr/lib/python2.7/site-packages/salt/client/ssh/wrapper/

**Comment by throwawayj** [comment]
My issues raised directly out of running salt commands from the master or salt-call from the minion. So if that is what salt-ssh is then yes.

On October 7, 2015, at 1:34 AM, Sascha Schuenemann <notifications@github.com> wrote:

@jfindlay yes, I have encountered this only with salt-ssh.

It seems that the calls to wfunc(s) (like pillar.*, saltutils.* ) are handled different than calls to custom modules. Our guess is that not only pillar data but other data like grains and mines might also be affected.

We were able to properly access pillar data using salt-ssh when placing a wrapper in /usr/lib/python2.7/site-packages/salt/client/ssh/wrapper/

---
